app/models/user.rb

```diff
@@ -52,9 +52,10 @@ def self.from_api_token(token)
     self.find_or_create_by_username_or_email(token.username, token.email)
   end
 
-  def self.find_for_identity(access_token, signed_in_resource=nil)
-    username = access_token.info['email']
-    User.find_by_username(username) || User.find_by_email(username) || User.create(username: username, email: username)
+  def self.find_for_generic(access_token, signed_in_resource=nil)
+    username = access_token.uid
+    email = access_token.info.email
+    User.find_by_username(username) || User.find_by_email(email) || User.create(username: username, email: email)
```

Comment:
Does this change cause problems because now the User instances' username and email aren't both username?

Reply:
I'm fine with just manually testing this.